src/language/mongoDBService.ts

```diff
@@ -181,4 +200,233 @@ export default class MongoDBService {
+      }
+
+      return resolve([]);
+    });
+  }
+
+  private checkHasDatabaseName(
+    node: any,
+    currentPosition: { line: number; character: number }
+  ): boolean {
+    if (
+      node.callee.name === 'use' &&
+      node.arguments &&
+      node.arguments.length === 1 &&
+      node.arguments[0].type === esprima.Syntax.Literal &&
+      (currentPosition.line >= node.loc.start.line ||
+        (currentPosition.line === node.loc.start.line - 1 &&
+          currentPosition.character > node.loc.end.column))
+    ) {
+      return true;
+    }
+
+    return false;
+  }
+
+  private checkHasCollectionName(
+    node: any,
+    currentPosition: { line: number; character: number }
+  ): boolean {
+    if (
+      node.object.name === 'db' &&
+      (currentPosition.line >= node.loc.start.line ||
+        (currentPosition.line === node.loc.start.line - 1 &&
+          currentPosition.character > node.loc.end.column))
+    ) {
+      return true;
+    }
+
+    return false;
+  }
+
+  private checkIsCurrentNode(
+    node: any,
+    currentPosition: { line: number; character: number }
+  ): boolean {
+    if (
+      currentPosition.line === node.loc.end.line - 1 &&
+      currentPosition.character === node.loc.end.column - 1
+    ) {
+      return true;
+    }
+
+    return false;
+  }
+
+  private parseAST(
+    text: string,
+    position: { line: number; character: number }
+  ): {
+    databaseName: string | null;
+    collectionName: string | null;
+    isObjectKey: boolean;
+    isMemberExpression: boolean;
+  } {
+    let databaseName = null;
+    let collectionName = null;
+    let isObjectKey = false;
+    let isMemberExpression = false;
+
+    try {
+      this._connection.console.log(
+        `ESPRIMA symbol position: ${util.inspect(position)}`
+      );
+      this._connection.console.log(`ESPRIMA completion body: "${text}"`);
+
+      const ast = esprima.parseScript(text, { loc: true });
+
+      estraverse.traverse(ast, {
+        enter: (node) => {
+          if (
+            node.type === esprima.Syntax.CallExpression &&
+            this.checkHasDatabaseName(node, position)
+          ) {
+            databaseName = node.arguments[0].value;
+          }
+
+          if (
+            node.type === esprima.Syntax.MemberExpression &&
+            this.checkHasCollectionName(node, position)
+          ) {
+            collectionName = node.property.name;
+
+            if (
+              this.checkIsCurrentNode(node, {
+                line: position.line,
+                character: position.character - 2
+              })
+            ) {
+              isMemberExpression = true;
+            }
+          }
+
+          if (
+            node.type === esprima.Syntax.ObjectExpression &&
+            this.checkIsCurrentNode(node, position)
+          ) {
+            isObjectKey = true;
+          }
+        }
+      });
+    } catch (error) {
+      this._connection.console.log(`ESPRIMA error: ${util.inspect(error)}`);
```

Comment:
Nice - I think it's important we're tolerant of errors and keep running - we're not quite sure what the users might have in their editor and it could be pretty error prone.

Reply:
Yeah we try to be very alive in any case.